skrub/_multi_agg_joiner.py

```diff
@@ -0,0 +1,499 @@
+                f" tables in `aux_tables`. Got {len(operations)} operations and"
+                f" {len(self._aux_tables)} auxiliary tables."
+            )
+        return operations
+
+    def _check_suffixes(self):
+        """Check that the len of `suffixes` match the len of `aux_tables`,
+        and that all suffixes are strings.
+
+        If suffixes is None, the suffixes will default to the position of
+        each auxiliary table in the list.
+
+        Returns
+        -------
+        suffixes
+            1-dimensional array of suffixes to append to dataframes.
+
+        Raises
+        ------
+        ValueError
+            If the len of `suffixes` doesn't match the len of `aux_tables`,
+            or if all suffixes are not strings.
+        """
+        if self.suffixes is None:
+            suffixes = [f"_{i}" for i in range(len(self._aux_tables))]
+        else:
+            suffixes = np.atleast_1d(self.suffixes).tolist()
+            if len(suffixes) != len(self._aux_tables):
+                raise ValueError(
+                    "The number of provided suffixes must match the number of"
+                    f" tables in `aux_tables`. Got {len(suffixes)} suffixes and"
+                    f" {len(self._aux_tables)} aux_tables."
+                )
+            if not all([isinstance(suffix, str) for suffix in suffixes]):
+                raise ValueError("All suffixes must be strings.")
+
+        return suffixes
+
+    def fit(self, X, y=None):
+        """Aggregate auxiliary tables based on the main keys.
+
+        Parameters
+        ----------
+        X : DataframeLike
+            Input data, based table on which to left join the
+            auxiliary tables.
+        y : None
+            Unused, only here for compatibility.
+
+        Returns
+        -------
+        MultiAggJoiner
+            Fitted :class:`MultiAggJoiner` instance (self).
+        """
+        X, self._aux_tables = self._check_dataframes(X, self.aux_tables)
+        self._main_keys, self._aux_keys = self._check_keys(
+            self.main_keys, self.aux_keys, self.keys, self._aux_tables
+        )
+        self._cols = self._check_cols()
+        self._operations = self._check_operations()
+        self._suffixes = self._check_suffixes()
+
+        self.agg_joiners_ = []
+        for aux_table, main_key, aux_key, cols, operations, suffix in zip(
+            self._aux_tables,
+            self._main_keys,
+            self._aux_keys,
+            self._cols,
+            self._operations,
+            self._suffixes,
+        ):
+            agg_joiner = AggJoiner(
+                aux_table=aux_table,
+                main_key=main_key,
+                aux_key=aux_key,
+                cols=cols,
+                operations=operations,
+                suffix=suffix,
+            )
+            self.agg_joiners_.append(agg_joiner)
+
+        for agg_joiner in self.agg_joiners_:
+            agg_joiner._check_inputs(X)
```

Comment:
It is, but I wanted to performed the checks separately in case there were a large number of operations done. My rationale was that I didn't want to perform a lot of operations without checking that all input data is correct, so it is indeed called here then in the fit

Reply:
If you think this is not necessary now / will be handled another way, I can remove it :)